aicssegmentation/workflow/workflow.py

```diff
@@ -162,3 +169,164 @@ def is_done(self) -> bool:
+
+    def is_valid_image(self, image_path: Path) -> bool:
+        """
+        Check if file at a given image_path and is a valid image type we support.
+
+        Params:
+            image_path (Path): image to check
+
+        Returns:
+            (bool): True if file has a supported file extension.
+        """
+        if not image_path.exists():
+            return False
+        if image_path.suffix.lower() in SUPPORTED_FILE_EXTENSIONS:
+            return True
+        else:
+            return False
+
+    def process_all(self):
+        """
+        Process all images in the input_dir with the workflow_definition used to set up the BatchWorkflow
+
+        Params:
+            none
+
+        Returns:
+            none
+        """
+
+        files = [f for f in self.input_path.glob("**/*") if f.is_file]
+        # Currently will save files in same format as they are in the input path
+        for f in files:
+            self._files_count += 1
+            if self.is_valid_image(f):
+                read_image = AICSImage(f)
+
+                try:
+                    # read and format image in the way we expect
+                    image_from_path = self.format_image_to_3d(read_image)
+                    # Run workflow on image
+                    workflow = Workflow(self._workflow_definition, image_from_path)
+                    result = workflow.execute_all()
+                    with OmeTiffWriter(self.output_path.joinpath(f.name), overwrite_file=True) as w:
+                        w.save(data=self.convert_bool_to_uint8(result), dimension_order="ZYX")
+
+                except Exception as e:
+                    # Handle failures during workflow execution/save
+                    self._failed_files += 1
+                    self.write_to_log_file(f"FAILED: {f}, ERROR: {e}\n")
+            else:
+                self._failed_files += 1
+                self.write_to_log_file(f"FAILED: {f}, ERROR: Unsupported Image Type {f.suffix}\n")
+        self._write_log_file_summary()
+
+    def _write_log_file_summary(self):
+        """
+        Write a log file to the output folder.
+
+        Params:
+            none
+
+        Returns:
+            none
+        """
+        if self._files_count == 0:
+            report = "There were no files to process in the input directory"
+        else:
+
+            files_processed = self._files_count - self._failed_files
+            report = f"{files_processed}/{self._files_count} files were processed.\n"
+        self.write_to_log_file(report + "\n")
+
+    def format_image_to_3d(self, image: AICSImage) -> np.ndarray:
+        """
+        Format images in the way that aics-segmention expects for most workflows (3d, zyx)
+
+        Params:
+            image_path (AICSImage): image to format
+
+        Returns:
+            np.ndarray: segment-able image for aics-segmentation
+        """
+        dims = len(image.shape) - image.shape.count(1)
+        if dims == 6:
```

Comment:
Note: I think this code can be simplified by using properties like AICSImage.size_s, AICSImage.size_t and AICSImage.size_c

Reply:
Note that the size_ properties are gone in aicsimageio 4.x and AICSImage.dims is the way to access dimensions by name.

Reply:
Good to know. I don't think we're quite ready to upgrade to 4.0 yet as it will probably have impacts on the whole repo, but hopefully we'll get that on the roadmap.